cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -281,8 +285,8 @@ public KafkaBrokerConfigurationBuilder withListeners(
         if (node.controller() || (node.broker() && kafkaMetadataConfigState.isZooKeeperToMigration())) {
             listeners.add(CONTROL_PLANE_LISTENER_NAME + "://0.0.0.0:9090");
 
-            // Control Plane listener to be advertised only with broker in ZooKeeper-based or migration but NOT when full KRaft only or mixed
-            if (node.broker() && kafkaMetadataConfigState.isZooKeeperToMigration()) {
+            // Control Plane listener to be advertised with broker in ZooKeeper-based or migration OR KRaft controller or mixed node with version 3.9+
+            if ((node.broker() && kafkaMetadataConfigState.isZooKeeperToMigration()) || (kafkaVersion.compareTo(KAFKA_VERSION_3_9) >= 0)) {
```

Comment:
I think you should just use here `KafkaVersion.compareDottedVersions(kafkaVersion.version(), "3.9.0") >= 0`. That is what we do in the other parts of the code. It will also mean that you won't need to create the KAFKA_VERSION_3_9 field:

```suggestion
            if ((node.broker() && kafkaMetadataConfigState.isZooKeeperToMigration()) || (KafkaVersion.compareDottedVersions(kafkaVersion.version(), "3.9.0") >= 0)) {
```